{CDAP-16353) Implement LocalScheduleFetcher to get schedule from local store

wyzhang opened the pull request:
Motiviation:
Having local vs remote impliation for ScheduleFetcher allows us to
fetcher schedules from local when using shared SQL as storage and
from remote when using non-sharing NOSQL.

We have already implemented RemoteScheduleFetcher. This change is
to introduce a local version.

Currently neither local nor remote is being used. A follow-up PR
will make metadata service use either local or remote depending on
storage implementation configuration.

yaojiefeng commented on cdap-app-fabric/src/main/java/io/cdap/cdap/metadata/LocalScheduleFetcher.java:
```diff
@@ -0,0 +1,78 @@
+      throw new IOException(e);
+    }
+    ScheduleDetail detail = record.toScheduleDetail();
+    return detail;
```
just return `record.toScheduleDetail`

wyzhang replied:
Done

yaojiefeng commented on cdap-app-fabric/src/main/java/io/cdap/cdap/metadata/LocalScheduleFetcher.java:
```diff
@@ -0,0 +1,78 @@
+        Throwables.propagateIfPossible(e.getCause(), IOException.class, ProgramNotFoundException.class);
+        throw new IOException(e);
+      }
+      List<ScheduleDetail> details = schedules.stream()
```
no need to create a variable, just return here

wyzhang replied:
Done.

yaojiefeng commented on cdap-app-fabric/src/test/java/io/cdap/cdap/metadata/ScheduleFetcherTest.java:
```diff
@@ -27,27 +27,56 @@
-  @BeforeClass
-  public static void setUp() {
-    fetcher = getInjector().getInstance(RemoteScheduleFetcher.class);
+  private ScheduleFetcherType fetcherType = null;
```
this can be private final, you don't need to assign null to it, since the constructor will assign value to it

wyzhang replied:
Done.

yaojiefeng reviewed (approved):
minor comments, rest LGTM

wyzhang committed:
{CDAP-16353) Implement LocalScheduleFetcher to get schedule from local store.

Motiviation:
Having local vs remote impliation for ScheduleFetcher allows us to
fetcher schedules from local when using shared SQL as storage and
from remote when using non-sharing NOSQL.

We have already implemented RemoteScheduleFetcher. This change is
to introduce a local version.

Currently neither local nor remote is being used. A follow-up PR
will make metadata service use either local or remote depending on
storage implementation configuration.
